frontend/src/app/modules/tenant-admin/client/client.component.ts

```diff
@@ -87,42 +93,57 @@ export class ClientComponent implements OnInit {
       saveData$ = this.apiService.updateCase(caseDetail, this.type);
     }
 
-    saveData$.subscribe(() => {
+    this.subs.sink = saveData$.subscribe(() => {
       this.snackbarService.success('Persönliche Daten erfolgreich aktualisiert');
       this.router.navigate(['/tenant-admin/clients']);
     });
   }
 
   startTracking(caseDetail: CaseDetailDto) {
-    this.apiService.putApiCall<StartTracking>(caseDetail, 'start-tracking')
+    this.subs.sink = this.apiService.putApiCall<StartTracking>(caseDetail, 'start-tracking')
       .subscribe((data) => {
         this.trackingStart$$.next(data);
         this.tabIndex = 3;
       });
   }
 
   renewTracking(tracking: HalResponse) {
-    this.apiService.putApiCall<StartTracking>(tracking, 'renew')
+    this.subs.sink = this.apiService.putApiCall<StartTracking>(tracking, 'renew')
       .subscribe((data) => {
         this.trackingStart$$.next(data);
         this.tabIndex = 3;
       });
   }
 
   addComment(commentText: string) {
-    this.apiService.addComment(this.caseId, commentText).subscribe((data) => {
+    this.subs.sink = this.apiService.addComment(this.caseId, commentText).subscribe((data) => {
       this.snackbarService.success('Kommentar erfolgreich eingetragen.');
       this.updatedDetail$$.next(data);
     });
   }
 
+  checkForClose(halResponse: HalResponse) {
+    this.subs.sink = this.matDialog.open(CloseCaseDialogComponent, {width: '640px'}).afterClosed().pipe(
+      switchMap((comment: string) => {
+        if (comment) {
+          return this.apiService.addComment(this.caseId, comment);
+        }
+        return of();
+      }),
+      map(() => this.closeCase(halResponse))
```

Comment:
vor switchMap() ein filter((comment) => !!comment), dann brauchst du das of() nicht ;-)